### DISCUSSION TITLE
Customize cells

### DISCUSSION THREAD

**Discussion by galegobr01** [pxlrbt/filament-excel#256] [Q&A]
Is there any way to customize, for example: Alignment, type of data entered (text, number, general), cell coloring, etc.?

**Comment by pxlrbt** [comment]
You probably need to extend the Exportable and add the methods from `laravel/excel`
https://docs.laravel-excel.com/3.1/exports/column-formatting.html#styling

**Comment by yukebrillianth** [comment]
> You probably need to extend the Exportable and add the methods from `laravel/excel` https://docs.laravel-excel.com/3.1/exports/column-formatting.html#styling

how to do it?

**Comment by Priet** [comment]
Maybe you figured it out already but I was looking for the same thing. For future reference, here is how I solved it.

First, create a custom export as explained here: https://github.com/pxlrbt/filament-excel?tab=readme-ov-file#custom-exports

```php
use pxlrbt\FilamentExcel\Actions\Tables\ExportAction;
use pxlrbt\FilamentExcel\Exports\ExcelExport;
use pxlrbt\FilamentExcel\Columns\Column;

class CustomExport extends ExcelExport
{
    
    public function setUp()
    {
        $this->withFilename('custom_export');
        $this->withColumns([
            Column::make('name'),
            Column::make('email'),
        ]);
    }
}
```

Then make sure the class implements the **WithStyles** interface:

```php
use Maatwebsite\Excel\Concerns\WithStyles;

class CustomExport extends ExcelExport implements WithStyles
{
    ....
```

At last, implement the public function **styles**:
```php
class CustomExport extends ExcelExport implements WithStyles
{
   ...

    public function styles(Worksheet $sheet)
    {
        return [
            1 => ['font' => ['bold' => true]],
            'D' => ['alignment' => ['horizontal' => 'right']],
        ];
    }
```

In this example, the header row will be bold and the 4th column will be right aligned.

Hope this helps!

**Comment by pxlrbt** [comment]
@Priet Thanks for sharing.

**Comment by synsyst** [comment]
@Priet is the styles also needed for excel formats? ie- dates become actual date formatted in the exported excel, numbers becomes numbers etc- not just general or strings

**Comment by pxlrbt** [comment]
@synsyst Late answer, but you can use `->columnFormats()` for that one.
